src/lib/stores/drips/utils/__test__/streamedBetween.test.ts

```diff
@@ -0,0 +1,220 @@
+import { flattenDripHistory } from '../streamedBetween';
+import { mockDripHistoryEvent, mockMoney } from './mockObjects';
+import { eth } from './utils';
+
+afterEach(() => {
+  jest.useRealTimers();
+});
+
+describe('flattenDripHistory', () => {
+  it('returns zero amounts if empty history is passed', () => {
+    expect(flattenDripHistory([])).toEqual({
+      amountStreamed: eth(0),
+      amountRemaining: eth(0)
+    });
+  });
+
+  it('correctly returns 1 dai as streamed amount for a stream that has streamed its entire balance of 1 dai', () => {
+    expect(
+      flattenDripHistory([
+        mockDripHistoryEvent({
+          balance: mockMoney(eth(1)),
+          amtPerSec: mockMoney(eth(0.1)),
+          seconds: 0
+        })
+      ])
+    ).toEqual({
+      amountStreamed: eth(1),
+      amountRemaining: eth(0)
+    });
+  });
+
+  it('takes time windows into account', () => {
+    expect(
+      flattenDripHistory(
+        [
+          mockDripHistoryEvent({
+            balance: mockMoney(eth(1)),
+            amtPerSec: mockMoney(eth(0.1)),
+            seconds: 0
+          })
+        ],
+        {
+          from: new Date(0 * 1000), // exactly the second the stream was set up
+          to: new Date(4 * 1000) // 4 seconds later
+        }
+      )
+    ).toEqual({
+      amountStreamed: eth(0.4),
+      amountRemaining: eth(0.6)
+    });
+
+    expect(
+      flattenDripHistory(
+        [
+          mockDripHistoryEvent({
+            balance: mockMoney(eth(1)),
+            amtPerSec: mockMoney(eth(0.1)),
+            seconds: 0
+          })
+        ],
+        {
+          from: new Date(0 * 1000), // exactly the second the stream was set up
+          to: new Date(9 * 1000) // 9 seconds later
+        }
+      )
+    ).toEqual({
+      amountStreamed: eth(0.9),
+      amountRemaining: eth(0.1)
+    });
+
+    expect(
+      flattenDripHistory(
+        [
+          mockDripHistoryEvent({
+            balance: mockMoney(eth(1)),
+            amtPerSec: mockMoney(eth(0.1)),
+            seconds: 0
+          })
+        ],
+        {
+          from: new Date(0 * 1000), // exactly the second the stream was set up
+          to: new Date(11 * 1000) // 11 seconds later
+        }
+      )
+    ).toEqual({
+      amountStreamed: eth(1),
+      amountRemaining: eth(0)
+    });
+  });
+
+  it('calculates the streamed amount from the beginning of time until the current date by default', () => {
```

Comment:
The test file would be easier to comprehend if this test was closer to the top, because I had to look at the code to understand what the default is.